### PR TITLE
[CBRD-24103] The permission of lob sub-directory and lob files have e…

### PR DESCRIPTION
**Pull request by eido5**
[http://jira.cubrid.org/browse/CBRD-24103](http://jira.cubrid.org/browse/CBRD-24103)

**Purpose**
The permission of lob sub-directory and lob files have execute permission. So it should be eliminated for the permission consistency and prevention the access.

**Implementation**
`ret = mkdir (dirbuf, 0755);` is changed to `ret = mkdir (dirbuf, 0744);` in `es_make_dirs()` function

**Remarks**
N/A

### CODE REVIEW

**Comment by wrlawodms** [comment]
@eido5 Please use the JIRA url format instead of the issue number at the head of the description.

**Comment by wrlawodms** [comment]
@eido5 I'd like to know the reason you change umask bits, which can affect other files, not each lob-related file permission
like `if (mkdir (lob_path, 0777) < 0)` in xboot_initialize_server().

**Review by wrlawodms — COMMENTED** [review]
@eido5 I'd like to know the reason you change umask bits, which can affect other files, not each lob-related file permission
like `if (mkdir (lob_path, 0777) < 0)` in `xboot_initialize_server()`.

File: src/transaction/log_manager.c [hunk 1]
```diff
@@ -839,7 +839,7 @@ log_create_internal (THREAD_ENTRY * thread_p, const char *db_fullname, const cha
    * Turn off creation bits for group and others
    */
 
-  (void) umask (S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH);
+  (void) umask (S_IRGRP | S_IWGRP | S_IXGRP | S_IROTH | S_IWOTH | S_IXOTH);
```
Comment (wrlawodms) [inline src/transaction/log_manager.c]: from `continuous-integration/appveyor/pr`, S_IXGRP and S_IXOTH doesn't seem to be defined on Windows.

**Comment by eido5** [comment]
> @eido5 Please use the JIRA url format instead of the issue number at the head of the description.

Thank you. I changed it.

**Comment by eido5** [comment]
> @eido5 I'd like to know the reason you change umask bits, which can affect other files, not each lob-related file permission like `if (mkdir (lob_path, 0777) < 0)` in `xboot_initialize_server()`.

1. If a user want to access the folders and files which are created after 'testdb_lgat' file is created, It can access the folder. So I think it might be a flaw

**Comment by eido5** [comment]
> @eido5 I'd like to know the reason you change umask bits, which can affect other files, not each lob-related file permission like `if (mkdir (lob_path, 0777) < 0)` in `xboot_initialize_server()`.

**Notation**

- umask(1p) : the utility what we can type in the command. ($ umask)
- umask(2) : the function what we can use in the source code.

As we know, "lob" sub-directory and "lob" files are affected umask(2) and it is written on the umask(1p). So, I decided to change the umask(2). 

(I eliminated the all permissions of "execute" to become permission consistency.)

Because when we change the umask(1p), the permission, between "lob" directory and "lob" sub-directory , "lob" files, are not consistent.

**Comment by wrlawodms** [comment]
@eido5 With 1p and 2, you must mean the manual page number like 1: User Commands, 2:System Calls. Is it correct? Regarding this, I still don't understand why you touch the umask bit instead of file permission for each creating lob directories and files. Is there something I misunderstand? 

As an example, I suppose that we can use `mkdir (lob_path, 0666)` to eliminate the "execute" permission.

File: src/transaction/log_manager.c (hunk 1 above)
Comment (eido5) [reply src/transaction/log_manager.c]: I verified this comments and I cannot understand why this error arises. 
At first, I will try to "push" git branch to test the only "umask (S_IRGRP)".

Because I think that this error can arises by additional bit permissions.

**Comment by eido5** [comment]
> @eido5 With 1p and 2, you must mean the manual page number like 1: User Commands, 2:System Calls. Is it correct? Regarding this, I still don't understand why you touch the umask bit instead of file permission for each creating lob directories and files. Is there something I misunderstand?

Yes, that's right.
I may guess my notation is insufficient to make you understand. I'm sorry.

At first, "lob" sub-directory is not created by `if (mkdir (lob_path, 0777) < 0)` in `xboot_initialize_server()`.
And it is created `ret = mkdir (dirbuf, 0755);` in `es_make_dirs()`
So, its permission is already written by umask system calls in `log_internal()`.

As you know above source code is located in `es_posix.c`.
So every directories are affected. 

And "lob" files are also affected similar reasons.

- `xes_posix_create_file()` -> `fd = open (new_path, O_WRONLY | O_CREAT | O_EXCL, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH | O_LARGEFILE);`

But I also should try to consider the way that not modifying all the files and directories permission.

**Comment by wrlawodms** [comment]
@eido5 Could you summarize the change of permissions for each file?
1. lob root dir: [AS-IS] -> [TO-BE]
2. lob dir for each row: [AS-IS] -> [TO-BE]
3. lob files in each lob dir: [AS-IS] -> [TO-BE]

**Comment by eido5** [comment]
> @eido5 Could you summarize the change of permissions for each file?

Of course.

1. lob root dir: N/A -> N/A
2. lob dir for each row: 711 -> 700
3. lob files in each lob dir: N/A -> N/A

File: src/storage/es_posix.c [hunk 2]
```diff
@@ -132,7 +132,7 @@ es_make_dirs (const char *dirname1, const char *dirname2)
       assert (false);
       return ER_ES_INVALID_PATH;
     }
-  ret = mkdir (dirbuf, 0755);
+  ret = mkdir (dirbuf, 0744);
```
Comment (hornetmj) [inline src/storage/es_posix.c:135]: The 'members of a group' and 'others' have read permission. I think that one of the reasons for this change is to protect our files created by the server regardless of the value of umask.

Comment (hornetmj) [reply src/storage/es_posix.c:135]: Ignore this comment. I forgot to set the umask(S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH) on log_initialize_internal() for a moment.

**Review by hornetmj — APPROVED** [review]

**Review by wrlawodms — COMMENTED** [review]
If there is a progress, please re-request review.

**Comment by eido5** [comment]
> If there is a progress, please re-request review.

Yes, I ask about the detailed specification to reporter. So the progress is delayed. 
I will request the re-request after there is the progress.

**Comment by eido5** [comment]
This is summary. (AS-IS -> TO-BE)

1. lob root dir: N/A -> 700
2. lob dir for each row: 711 -> 700
3. lob files in each lob dir: N/A -> N/A

**Review by hornetmj — APPROVED** [review]

**Comment by wrlawodms** [comment]
@eido5 What is the feedback from the reporter? And please give us the AS-IS permission too (not N/A) in the summary.

**Comment by eido5** [comment]
> @eido5 What is the feedback from the reporter? And please give us the AS-IS permission too (not N/A) in the summary.

The reporter requested the totally consistent permission from the lob directory to the lob files because the government suggested that no execute permission is allowed.
But we know it is impossible because the permission of execute provides the entrance of the directory(ex directory permission 700 -> owner can enter the directory).

So we compromised that 
(AS-IS -> TO-BE)

1. lob root dir: 777 -> 700
2. lob dir for each row: 711 -> 700
3. lob files in each lob dir: 600 -> 600

**Review by wrlawodms — APPROVED** [review]

**Review by joohok — APPROVED** [review]
